### PR TITLE
Mappings improvements

### PR DESCRIPTION
**Pull request by sergi-unity**
Purpose of this PR
This PR is a refactor of everything related to BlendShapeMappings.
BlendShapesMappings asset now stores explicit bindings for each SkinnedMeshRenderer.
BlendShapesMappingsEditor allows to quickly build the data from a prefab and match automatically the blend-shape locations.
BlendShapesController now uses the mappings asset more efficiently without needing to process strings or maintain an internal list of BlendShapeIndices.

Testing status
Tested manually with Sloth scene and windup girl demo assets.

Documentation status
No documentation

Technical risk
Low.
Halo effect
Low.

### CODE REVIEW

File: Runtime/Scripts/BlendShapeMappings.cs [hunk 1]
```diff
@@ -1,75 +1,107 @@
...
-                {
-                    maps.Add("");
-                }
+#if UNITY_EDITOR
+        [SerializeField]
+        GameObject m_Prefab;
 
-                m_BlendShapeNames = maps.ToArray();
-            }
+        public GameObject prefab
```
Comment (bradweiers) [inline Runtime/Scripts/BlendShapeMappings.cs]: Add comments to public properties

File: Runtime/Scripts/BlendShapeMappings.cs [hunk 2]
```diff
@@ -1,75 +1,107 @@
...
+    {
+        [SerializeField]
+        List<BlendShapeMap> m_Maps = new List<BlendShapeMap>();
 
-                for (var i = maps.Count; i < m_LocationIdentifiers.Length; i++)
-                {
-                    maps.Add("");
-                }
+#if UNITY_EDITOR
+        [SerializeField]
```
Comment (bradweiers) [inline Runtime/Scripts/BlendShapeMappings.cs]: Add tooltip

File: Runtime/Scripts/BlendShapeMappings.cs [hunk 3]
```diff
@@ -1,75 +1,107 @@
...
+        }
+    }
+
+    /// <summary>
+    /// Asset storing a collection of BlendShapeMaps for a specific rig
+    /// </summary>
+    [CreateAssetMenu(fileName = "BlendShape Mappings", menuName = "AR Face Capture/BlendShape Mappings")]
+    public class BlendShapeMappings : ScriptableObject
+    {
+        [SerializeField]
```
Comment (bradweiers) [inline Runtime/Scripts/BlendShapeMappings.cs]: Add tooltip

File: Runtime/Scripts/BlendShapeOverride.cs [hunk 4]
```diff
@@ -87,9 +87,9 @@ public float blendShapeOffset
 
         BlendShapeOverride(){}
 
-        public BlendShapeOverride(string name)
+        public BlendShapeOverride(BlendShapeLocation location)
```
Comment (bradweiers) [inline Runtime/Scripts/BlendShapeOverride.cs]: Comment all public methods

File: Runtime/Scripts/BlendShapeValues.cs [hunk 5]
```diff
@@ -120,7 +121,7 @@ public struct BlendShapeValues
         [FieldOffset(200)] public float NoseSneerRight;
         [FieldOffset(204)] public float TongueOut;
 
-        public int Count
+        public static int Count
```
Comment (bradweiers) [inline Runtime/Scripts/BlendShapeValues.cs]: Public properties should be lower-case.

File: Runtime/Scripts/BlendShapeMappings.cs (hunk 3 above)
Comment (sergi-unity) [reply Runtime/Scripts/BlendShapeMappings.cs]: I'll use HideInInspector for this one.

File: Editor/LevenshteinDistance.cs [hunk 6]
```diff
@@ -0,0 +1,58 @@
+﻿using System;
+
+namespace Unity.Labs.FacialRemote
+{
+    /// <summary>
+    /// Contains approximate string matching
```
[the remark this answers is not in the document]
Comment (sergi-unity) [reply Editor/LevenshteinDistance.cs]: Fixed in latest commit

File: Editor/LevenshteinDistance.cs [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    /// </summary>
+    internal static class LevenshteinDistance
+    {
+        /// <summary>
+        /// Compute the distance between two strings.
+        /// </summary>
+        public static int Compute(string s, string t)
+        {
+        int n = s.Length;
```
[the remark this answers is not in the document]
Comment (sergi-unity) [reply Editor/LevenshteinDistance.cs]: Fixed in latest commit

File: Runtime/Scripts/BlendShapeMappings.cs (hunk 1 above)
Comment (sergi-unity) [reply Runtime/Scripts/BlendShapeMappings.cs]: Fixed in latest commit

File: Runtime/Scripts/BlendShapeMappings.cs (hunk 2 above)
Comment (sergi-unity) [reply Runtime/Scripts/BlendShapeMappings.cs]: Fixed in latest commit

File: Runtime/Scripts/BlendShapeMappings.cs (hunk 3 above)
Comment (sergi-unity) [reply Runtime/Scripts/BlendShapeMappings.cs]: Fixed in latest commit

File: Runtime/Scripts/BlendShapeOverride.cs (hunk 4 above)
Comment (sergi-unity) [reply Runtime/Scripts/BlendShapeOverride.cs]: Fixed in latest commit

File: Runtime/Scripts/BlendShapeValues.cs (hunk 5 above)
Comment (sergi-unity) [reply Runtime/Scripts/BlendShapeValues.cs]: Fixed in latest commit

**Comment by sergi-unity** [comment]
Added BlendShapeMappings instructions to the readme file

**Pull request merged**